packages/mermaid/src/rendering-util/selectSvgElement.ts

```diff
@@ -0,0 +1,24 @@
+import { select } from 'd3';
+import { getConfig } from '../config.js';
+import type { HTML, SVG } from '../diagram-api/types.js';
+
+/**
+ * Selects the SVG element using {@link id}.
+ *
+ * @param id - The diagram ID.
+ * @returns The selected {@link SVG} element using {@link id}.
+ */
+export const selectSvgElement = (id: string): SVG => {
+  const { securityLevel } = getConfig();
+  // handle root and document for when rendering in sandbox mode
+  let doc: Document = document;
+  if (securityLevel === 'sandbox') {
+    const sandboxElement: HTML = select(`#i${id}`);
+    doc = sandboxElement.node()?.contentDocument ?? doc;
+  }
+  const root: HTML =
+    securityLevel === 'sandbox' ? select(doc.body as HTMLIFrameElement) : select('body');
```

Comment:
Seems, that we refer to `doc` variable only with 'sandbox' security level, so:
```ts
const root: HML = select('body');
if (securityLevel === 'sandbox') {
    const sandboxElement: HTML = select(`#i${id}`);
    const doc: Document = sandboxElement.node()?.contentDocument ?? document;
    root = select(doc.body as HTMLIFrameElement);
}
```
Unless we need to return `doc` along with `svg` as a function result

Reply:
And a separate comment for this. Is there need to return `root`, `doc` along with `svg` for other diagrams? Info diagram is our lab animal in many cases, but not fully representative.

Reply:
> Seems, that we refer to `doc` variable only with 'sandbox' security level, so:
> 
> ```ts
> const root: HML = select('body');
> if (securityLevel === 'sandbox') {
>     const sandboxElement: HTML = select(`#i${id}`);
>     const doc: Document = sandboxElement.node()?.contentDocument ?? document;
>     root = select(doc.body as HTMLIFrameElement);
> }
> ```
> 
> Unless we need to return `doc` along with `svg` as a function result

Make sense, thanks!

> And a separate comment for this. Is there need to return `root`, `doc` along with `svg` for other diagrams? Info diagram is our lab animal in many cases, but not fully representative.

And don't think we need to do that. While I was working on converting pie files, I saw it [uses the `doc`](https://github.com/mermaid-js/mermaid/blob/23b6d53f80fb803562f13f86039da31a1cf632dd/packages/mermaid/src/diagrams/pie/pieRenderer.ts#L36) and [change some attributes](https://github.com/mermaid-js/mermaid/blob/23b6d53f80fb803562f13f86039da31a1cf632dd/packages/mermaid/src/diagrams/pie/pieRenderer.ts#L44) from it, on it, which it should do that on the `svg` instead.
Some diagrams do that and I'll fix them when standardizing their files.